### ISSUE TITLE
Update code to perform multiple runs of model training and select model with best cross-val loss

### ISSUE THREAD

**Issue by abuzarmahmood** [pulakatlab/auto_slide#51]
look in pipeline/model/training.py

**Comment by abuzarmahmood** [comment]
Related evidence, tracked under #116.

Two reproducibility problems in the current split that are worth folding into
this work:

**The split is unseeded and redrawn every run.**

https://github.com/pulakatlab/autoslide/blob/main/autoslide/src/pipeline/model/data_preprocessing.py#L81

```python
train_imgs_inds = np.random.choice(range(len(image_names)), num, replace=False)
```

No seed, and the split is not persisted, so there is no record of which images a
given checkpoint was trained on.

**The augmented set is cached on disk across splits.**
`load_or_create_augmented_data` skips regeneration entirely if
`augmented_images/` already exists:

```python
if os.path.exists(aug_img_dir) and len(os.listdir(aug_img_dir)) > 0:
    print("Augmented images already exist. Skipping augmentation...")
```

Those cached files were derived from whatever split was in effect when they were
first generated. On any later run with a fresh random split, augmented crops
derived from images that are now in validation get used as training data.

Suggested, alongside the multi-run selection this issue asks for:

- seed the split and write the resulting file lists next to the checkpoint;
- key the augmentation cache to the split (or regenerate it per run), and note
  that the current `augmented_images/` directory cannot be assumed consistent
  with any particular checkpoint;
- for the cross-val selection itself, hold the folds fixed across runs so the
  comparison is like-for-like.
